run_all_tests.sh

```
@@ -0,0 +1,64 @@
+    echo -e "${BLUE}========================================${NC}"
+
+    # 1. Configuration
+    echo -e "${BLUE}[1/3] Configuring $mode...${NC}"
+    cmake -S "$PROJECT_ROOT" -B "$build_dir" \
+          -DCMAKE_BUILD_TYPE="$mode" \
+          -DBUILD_TESTS=ON \
+          -DBUILD_EXAMPLES=OFF || report_failure "Configuration failed for $mode mode."
+
+    # 2. Compilation
+    echo -e "${BLUE}[2/3] Compiling $mode...${NC}"
+    cmake --build "$build_dir" -j$(nproc) || report_failure "Compilation failed for $mode mode."
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)